### ISSUE TITLE
Questions about train.py and test.py

### ISSUE THREAD

**Issue by BugMaker2002** [Dylan-H-Wang/SLF-RPM#13]
Hello, can you tell me what this code does in the test.py file?（您好，请问您能否告诉我在test.py文件当中这段代码的作用是什么？）
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/81f7910e-b1ea-445f-a5c2-7d3e35fc439e)
And why are the model architectures used in train.py and test.py different? In train.py, you used a network architecture of ndim=2048, but in test.py, you seem to have changed to a network architecture of n_class=1, so if you load the pre-trained model directly, you will get an error because the dimensions of the parameters do not match. I think the normal approach would be to use the same network architecture on the training set and the validation set, the model on the validation set loads the trained model weights directly on the training set, and then we train an MLP with nn.Linear(2048, 1), but why don't you do that?（而且为什么在train.py和test.py当中使用的模型架构也不同？在train.py文件当中，您使用的是ndim=2048的网络架构，但是在test.py文件当中，您似乎又换成了n_class=1的网络架构，这样的话如果直接加载预训练模型是会报错的，因为参数的维度对应不上。我觉得正常的做法应该是：在训练集和验证集上使用相同的网络架构，验证集上的模型直接载入训练集训练好的模型权重，然后我们再去训练一个nn.Linear(2048, 1)的MLP，但是为什么您不选择这么做呢？）
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/d928eaba-795b-4ab2-88b3-affe1312acb9)

**Comment by Dylan-H-Wang** [comment]
Since the model was trained using DDP, the model is wrapped with `module`. We need to remove this prefix before loading the pre-trained model weights.

The difference between train and test model is a common practice in SSL (see SimCLR or MoCo for details). Their backbones are the same and the subsequent MLP part is different. That is why we discard `encoder_q.fc` when loading the pre-trained weights. For fine-tuning the model on downstream tasks, we discard the MLP after backbone and append a new MLP and train it on the dataset.

**Comment by BugMaker2002** [comment]
But I printed out all the keys in state_dict and found that there was no part beginning with `"module"`, that is, the first if statement: `if k.S. tartswith("module.encoder_q") and not k.s. tartswith("module.encoder_q.fc"):` will not be executed, does that mean this statement is redundant? Also, what does DDP stand for?Thank you very much for answering my questions
![image](https://github.com/Dylan-H-Wang/SLF-RPM/assets/110211332/2d8569eb-d279-4ed7-8cfc-306ad688ec7a)

**Comment by Dylan-H-Wang** [comment]
It is not redundant, but useful to have a sanity check.

You can google by yourself about PyTorch DDP.

**Comment by BugMaker2002** [comment]
Thank you very much
